### PR TITLE
integration tests for the invite function

### PR DESCRIPTION
**Pull request by individual-it**
This depends on #61 and https://github.com/owncloud/ocis/pull/7687

How to run:
1. build a docker image of the branch from https://github.com/owncloud/ocis/pull/7687
    e.g.
    ```
    make -C ocis release-linux-docker-amd64
    docker build --rm=true -f ocis/docker/Dockerfile.linux.amd64 -t ocis:sharing ocis --pull=true
    ```
2. set the name of the image in `services.ocis.image` inside of `tests/integration/compose.yaml`
3. run all integration tests `make test-php-integration`
4. or run only invite tests:
    ```
    make run-ocis-with-keycloak
    vendor/bin/phpunit --configuration ./phpunit.xml --testsuite integration --filter testInvite
    ```

### CODE REVIEW

**Comment by phil-davis** [comment]
@individual-it this PR branch needs to be rebased.

File: tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[1]->getName());
+    }
+
+    public function testInviteSameUserAgain(): void
+    {
+        $this->markTestSkipped('https://github.com/owncloud/ocis/issues/7842');
+        // @phpstan-ignore-next-line because the test is skipped
+        $this->expectException(ForbiddenException::class);
+        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
+        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
```
Comment (amrita-shrestha) [inline tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php:134]: how about using anotherRole for second invite
```suggestion
        $this->resourceToShare->invite([$this->einstein], $this->viewerRole);
        $this->resourceToShare->invite([$this->einstein], $this->anotherRole);
```

File: tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        $this->createdGroups = [$philosophyHatersGroup];
+        $philosophyHatersGroup->addUser($this->einstein);
+        $shares = $this->resourceToShare->invite([$philosophyHatersGroup], $this->viewerRole);
+        $this->assertCount(1, $shares);
+        $receivedShares = $this->einsteinOcis->getSharedWithMe();
+        $this->assertCount(1, $receivedShares);
+        $this->assertSame($this->resourceToShare->getName(), $receivedShares[0]->getName());
+    }
+
+    public function testInviteGroupAndUserOfTheGroup(): void
```
Comment (amrita-shrestha) [inline tests/integration/Owncloud/OcisPhpSdk/ResourceInviteTest.php:112]: testInviteMultipleGroupWithCommonUser can be added too

**Comment by individual-it** [comment]
I've added two more tests. Generally we have to find the right balance of testing here, we want to test all the code of the SDK including all edge cases we can think of, but on the other hand we don't need to test ocis in our tests. Testing if the API of ocis behaves correctly is part of the tests in ocis

**Review by amrita-shrestha — APPROVED** [review]
lgtm
